--- Context.py	2018-08-09 02:47:32.000000000 +0200
+++ Context.py	2024-10-05 21:08:01.642240039 +0200
@@ -2,7 +2,8 @@
 # encoding: utf-8
 # WARNING! Do not edit! https://waf.io/book/index.html#_obtaining_the_waf_file
 
-import os,re,imp,sys
+#import os,re,imp,sys
+import os,re,importlib.util,sys
 from waflib import Utils,Errors,Logs
 import waflib.Node
 HEXVERSION=0x2000a00
@@ -15,6 +16,7 @@
 TOP='top'
 OUT='out'
 WSCRIPT_FILE='wscript'
+module_name = WSCRIPT_FILE
 launch_dir=''
 run_dir=''
 top_dir=''
@@ -106,7 +108,7 @@
 				cache[node]=True
 				self.pre_recurse(node)
 				try:
-					function_code=node.read('rU',encoding)
+					function_code=node.read('r',encoding)
 					exec(compile(function_code,node.abspath(),'exec'),self.exec_dict)
 				finally:
 					self.post_recurse(node)
@@ -344,9 +346,11 @@
 		return cache_modules[path]
 	except KeyError:
 		pass
-	module=imp.new_module(WSCRIPT_FILE)
+#	module=imp.new_module(WSCRIPT_FILE)
+	module = importlib.util.module_from_spec(importlib.util.spec_from_loader(module_name, loader=None))
+	sys.modules[module_name] = module
 	try:
-		code=Utils.readf(path,m='rU',encoding=encoding)
+		code=Utils.readf(path,m='r',encoding=encoding)
 	except EnvironmentError:
 		raise Errors.WafError('Could not read the file %r'%path)
 	module_dir=os.path.dirname(path)
