diff --git a/common/lwlibav_video.c b/common/lwlibav_video.c
index bf0e859..50a5169 100644
--- a/common/lwlibav_video.c
+++ b/common/lwlibav_video.c
@@ -362,16 +362,8 @@ use_lavf_frame_rate:
     }
     else
     {
-        if (lwhp->raw_demuxer)
-        {
-            *framerate_num = (int64_t)stream->codecpar->framerate.num;
-            *framerate_den = (int64_t)stream->codecpar->framerate.den;
-        }
-        else
-        {
-            *framerate_num = (int64_t)stream->avg_frame_rate.num;
-            *framerate_den = (int64_t)stream->avg_frame_rate.den;
-        }        
+         *framerate_num = (int64_t)stream->avg_frame_rate.num;
+         *framerate_den = (int64_t)stream->avg_frame_rate.den;
     }
     return;
 }
