--- lwlibav_video.c.ori	2023-12-20 22:44:08.387842399 +0100
+++ lwlibav_video.c.mod	2023-12-20 22:41:05.225003527 +0100
@@ -362,16 +362,8 @@
     }
     else
     {
-        if (vdhp->format->iformat->raw_codec_id)
-        {
-            *framerate_num = (int64_t)stream->codecpar->framerate.num;
-            *framerate_den = (int64_t)stream->codecpar->framerate.den;
-        }
-        else
-        {
-            *framerate_num = (int64_t)stream->avg_frame_rate.num;
-            *framerate_den = (int64_t)stream->avg_frame_rate.den;
-        }        
+        *framerate_num = (int64_t)stream->avg_frame_rate.num;
+        *framerate_den = (int64_t)stream->avg_frame_rate.den;
     }
     return;
 }
